bin/newyear.pl

```diff
@@ -412,26 +429,18 @@ sub SubmitNewYearReview
   if ($result) {
     print RED "SubmitReview() for $id: $result\n";
   }
-  if ($excel) {
+  if ($tsv) {
     my $rq = $crms->RightsQuery($id, 1);
     my ($acurr, $rcurr, $src, $usr, $timecurr, $note) = @{$rq->[0]};
-    $worksheet->write_string($wsrow, 0, $id);
-    $worksheet->write_string($wsrow, 1, $project_name);
-    $worksheet->write_string($wsrow, 2, $record->author || '');
-    $worksheet->write_string($wsrow, 3, $record->title || '');
-    $worksheet->write_string($wsrow, 4, $record->copyrightDate || '');
-    $worksheet->write_string($wsrow, 5, $record->country || '');
-    $worksheet->write_string($wsrow, 6, "$acurr/$rcurr");
-    $worksheet->write_string($wsrow, 7, $extracted_data);
-    $worksheet->write_string($wsrow, 8, $predictions);
-    $worksheet->write_string($wsrow, 9, "$new_attr/$new_reason");
-    $worksheet->write_string($wsrow, 10, $msg);
-    print GREEN "Worksheet row $wsrow written\n";
-    $wsrow++;
+    my @values = ($id, $project_name, $record->author || '',
+      $record->title || '', $record->copyrightDate || '',
+      $record->country || '', "$acurr/$rcurr", $extracted_data,
+      $predictions, "$new_attr/$new_reason", $msg);
+    @values = map { local $_ = $_; $_ =~ s/\s+/ /g; $_; } @values;
```

Comment:
Is this different from `map { s/\s+/ /g; } @values`? Is it just that it (essentially) makes a copy rather than modifying it in-place? Is there any reason for that given that `@values` is local anyway?

Reply:
It's something to do with the way `s///` returns the number of replacements (or whether a replacement was made, some Perlism like that) rather than the operand. I found it counterintuitive to do it this way but the point was to just make sure that the TSV values don't have any residual whitespace noise.